xarray/coding/cftimeindex.py

```diff
@@ -359,13 +359,26 @@ def shift(self, n, freq):
                 "str or datetime.timedelta, got {}.".format(freq))
 
     def __add__(self, other):
+        if isinstance(other, pd.TimedeltaIndex):
+            other = other.to_pytimedelta()
         return CFTimeIndex(np.array(self) + other)
 
     def __radd__(self, other):
+        if isinstance(other, pd.TimedeltaIndex):
+            other = other.to_pytimedelta()
         return CFTimeIndex(other + np.array(self))
 
     def __sub__(self, other):
-        return CFTimeIndex(np.array(self) - other)
+        if isinstance(other, CFTimeIndex):
+            return pd.TimedeltaIndex(np.array(self) - np.array(other))
+        elif isinstance(other, pd.TimedeltaIndex):
+            return CFTimeIndex(np.array(self) - other.to_pytimedelta())
+        else:
+            return CFTimeIndex(np.array(self) - other)
+
+    def _add_delta(self, deltas):
+        # To support TimedeltaIndex + CFTimeIndex with older pandas versions
```

Comment:
Has this been changed now? As you know, it's slightly dangerous to dip into pandas internals...

Reply:
Yes, this method is not needed with recent versions of pandas.  I think the change that eliminated the need for it was in this PR: https://github.com/pandas-dev/pandas/pull/19895, which I guess would have made it into pandas version 0.23.0.  

I added it because of the failing test in the Windows 32-bit Python 2.7 build that used pandas version 0.22.0.  Should I just mark the test as an expected failure for versions of pandas less than 0.23 and remove this private method?